phpstan.neon.dist

```diff
@@ -96,9 +96,19 @@ parameters:
             paths:
                 - lib/Doctrine/DBAL/Connection.php
 
-        # https://github.com/phpstan/phpstan/issues/5608
+        # PHPStan fails to parse this type alias which is meant for Psalm only.
         -
-            message: '~^Circular definition detected in type alias (Override)?Params\.$~'
+            message: '~^Invalid type definition detected in type alias (Override)?Params\.$~'
+            paths:
+                - lib/Doctrine/DBAL/DriverManager.php
+
+        -
+            message: '~Template type T of method Doctrine\\DBAL\\DriverManager::getConnection\(\) is not referenced in a parameter\.~'
+            paths:
+                - lib/Doctrine/DBAL/DriverManager.php
+
+        -
+            message: '~Method Doctrine\\DBAL\\DriverManager::createDriver\(\) should return Doctrine\\DBAL\\Driver but returns object\.~'
```

Comment:
Should we provide some PHPStan-friendly annotation instead? Not sure how feasible this is though.

Reply:
In this file, we declare two type aliases for associative arrays, `Params` and `OverrideParams`. PHPStan flags those aliases as invalid, see the ignored error on line 101. The other two errors are a consequence of PHPStan being unaware of the array structure. `createDriver()` for instance contains the statement `return new $params['driverClass']();`.

However, if I remove `driver?: key-of<self::DRIVER_MAP>` from both aliases, PHPStan parses the aliases. It is probably possible to make this work somehow by defining versions of those aliases tailored for PHPStan, but I don't think that it's worth it. Psalm should have us covered here.